src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_collect.c

```diff
@@ -1,103 +1,1003 @@
+    255, // QUEST_REMAINS_GYORG
+    255, // QUEST_REMAINS_TWINMOLD
+    255, // Unused
+    255, // Unused
+};
+s16 sQuestRemainsEnvGreen[] = {
+    255, // QUEST_REMAINS_ODOWLA
+    255, // QUEST_REMAINS_GOHT
+    255, // QUEST_REMAINS_GYORG
+    255, // QUEST_REMAINS_TWINMOLD
+    255, // Unused
+    255, // Unused
+};
+s16 sQuestRemainsEnvBlue[] = {
+    150, // QUEST_REMAINS_ODOWLA
+    150, // QUEST_REMAINS_GOHT
+    150, // QUEST_REMAINS_GYORG
+    150, // QUEST_REMAINS_TWINMOLD
+    150, // Unused
+    150, // Unused
+};
 
-s32 D_8082AEE0 = 0;
+#ifdef NON_EQUIVALENT
+// TODO: is actually NON_MATCHING, update once `z_kaleido_scope_NES.c` is decompiled
+// A single small regalloc at the first `func_8010DC58` for the heart piece count (see `gItemIcons`)
+void KaleidoScope_DrawQuestStatus(PlayState* play) {
+    static s16 sQuestRemainsColorTimer = 20;
+    static s16 sQuestRemainsColorTimerIndex = 0;
+    static s16 sQuestHpPrimRed = 0;
+    static s16 sQuestHpPrimGreen = 0;
+    static s16 sQuestHpPrimBlue = 0;
+    static s16 sQuestHpPrimAlpha = 0;
+    static s16 sQuestHpColorTimer = 20;
+    static s16 sQuestHpPrimColorTargetIndex = 0;
+    static TexturePtr sOcarinaButtonTextures[] = {
+        gOcarinaATex, gOcarinaCDownTex, gOcarinaCRightTex, gOcarinaCLeftTex, gOcarinaCUpTex,
+    };
+    static s16 sQuestSongsPrimRed[] = {
+        150, // QUEST_SONG_SONATA
+        255, // QUEST_SONG_LULLABY
+        100, // QUEST_SONG_BOSSA_NOVA
+        255, // QUEST_SONG_ELEGY
+        255, // QUEST_SONG_OATH
+        255, // QUEST_SONG_SARIA
+        255, // QUEST_SONG_TIME
+        255, // QUEST_SONG_HEALING
+        255, // QUEST_SONG_EPONA
+        255, // QUEST_SONG_SOARING
+        255, // QUEST_SONG_STORMS
+        255, // QUEST_SONG_SUN
+    };
+    static s16 sQuestSongsPrimGreen[] = {
+        255, // QUEST_SONG_SONATA
+        80,  // QUEST_SONG_LULLABY
+        150, // QUEST_SONG_BOSSA_NOVA
+        160, // QUEST_SONG_ELEGY
+        100, // QUEST_SONG_OATH
+        240, // QUEST_SONG_SARIA
+        255, // QUEST_SONG_TIME
+        255, // QUEST_SONG_HEALING
+        255, // QUEST_SONG_EPONA
+        255, // QUEST_SONG_SOARING
+        255, // QUEST_SONG_STORMS
+        255, // QUEST_SONG_SUN
+    };
+    static s16 sQuestSongsPrimBlue[] = {
+        100, // QUEST_SONG_SONATA
+        40,  // QUEST_SONG_LULLABY
+        255, // QUEST_SONG_BOSSA_NOVA
+        0,   // QUEST_SONG_ELEGY
+        255, // QUEST_SONG_OATH
+        100, // QUEST_SONG_SARIA
+        255, // QUEST_SONG_TIME
+        255, // QUEST_SONG_HEALING
+        255, // QUEST_SONG_EPONA
+        255, // QUEST_SONG_SOARING
+        255, // QUEST_SONG_STORMS
+        255, // QUEST_SONG_SUN
+    };
+    static TexturePtr sQuestUpgradeTextures[][3] = {
+        { 0x08053000, 0x08054000, 0x08055000 }, // UPG_QUIVER
+        { 0x08056000, 0x08057000, 0x08058000 }, // UPG_BOMB_BAG
+    };
+    static u8 sQuestUpgrades[] = { UPG_QUIVER, UPG_BOMB_BAG };
```

Comment:
I have to ask: all needed in the function?

Reply:
Unfortunately yes. All those color pulsations tend to use data, and those all tend to be directly affected by in-function data. So `sQuestRemainsColorTimer` and those color temps at the top need to be in-function, meaning everything below does as well

Reply:
😮‍💨